test/EndToEndTests/Tests/Client/Microsoft.OData.Client.E2E.Tests/ActionOverloadingTests/Server/ProductsController.cs

```diff
@@ -0,0 +1,56 @@
+﻿//-----------------------------------------------------------------------------
+// <copyright file="ProductsController.cs" company=".NET Foundation">
+//      Copyright (c) .NET Foundation and Contributors. All rights reserved.
+//      See License.txt in the project root for license information.
+// </copyright>
+//------------------------------------------------------------------------------
+
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.OData.Formatter;
+using Microsoft.AspNetCore.OData.Query;
+using Microsoft.AspNetCore.OData.Routing.Controllers;
+using Microsoft.OData.Client.E2E.Tests.Common.Server.EndToEnd;
+
+namespace Microsoft.OData.Client.E2E.Tests.ActionOverloadingTests.Server
+{
+    public class ProductsController : ODataController
+    {
+        public ProductsController()
+        {
+        }
+
+        [EnableQuery]
+        public IActionResult Get()
+        {
+            var products = CommonEndToEndDataSource.Products;
+            return Ok(products);
+        }
+
+        [EnableQuery]
+        public IActionResult Get(int key)
+        {
+            var product = CommonEndToEndDataSource.Products?.FirstOrDefault(a => a.ProductId == key);

```

Comment:
What if the product doesn't exist, why not return a 404? Also, why does `Products` have the `?` operator here?

Reply:
Why not return a 404 if the entity is not found?

Reply:
fixed